test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/KeepAliveTimeoutTests.cs

```diff
@@ -50,17 +49,7 @@ public class KeepAliveTimeoutTests
                     "",
                     "");
                 await ReceiveResponse(connection, server.Context);
-
-                await Task.Delay(LongDelay);
-
-                await Assert.ThrowsAsync<IOException>(async () =>
-                {
-                    await connection.Send(
-                        "GET / HTTP/1.1",
-                        "",
-                        "");
-                    await ReceiveResponse(connection, server.Context);
-                });
+                await connection.WaitForConnectionClose().TimeoutAfter(LongDelay);
```

Comment:
`TestConnection.Receive` would randomly hang without a `Send` first. I think it has to do with socket state. Actually when I added `Send`, it would throw there and never get to the subsequent `Receive`. I could have removed the second receive. But just waiting for the FIN is way cleaner anyways.